packaging/requirements.py

```diff
@@ -31,12 +32,13 @@ class Requirement:
     # TODO: Can we normalize the name and extra name?
 
     def __init__(self, requirement_string: str) -> None:
+        _RequirementTuple = namedtuple(
+            "_RequirementTuple", ["name", "url", "extras", "specifier", "marker"]
+        )
```

Comment:
This should be computed outside of `__init__` to avoid paying the cost of creating this class each time an instance is created.